packages/components/src/components/text/text.jsx

```diff
@@ -1,15 +1,28 @@
 import classNames from 'classnames';
 import React from 'react';
 import PropTypes from 'prop-types';
+import { isEmptyObject } from '@deriv/shared';
 
-const Text = ({ children, size, color, align, weight, line_height, as, className, ...props }) => {
-    const style = {
-        '--text-size': `var(--text-size-${size || 's'})`,
-        '--text-color': `var(--text-${color || 'general'})`,
-        '--text-lh': `var(--text-lh-${line_height || 'm'})`,
-        '--text-weight': `var(--text-weight-${weight || 'normal'})`,
+const Text = ({ children, size, color, align, weight, line_height, as, className, styles, ...props }) => {
+    const setStyle = () => {
+        const class_styles = {
+            '--text-size': `var(--text-size-${size || 's'})`,
+            '--text-color': `var(--text-${color || 'general'})`,
+            '--text-lh': `var(--text-lh-${line_height || 'm'})`,
+            '--text-weight': `var(--text-weight-${weight || 'normal'})`,
+        };
+        if (!isEmptyObject(styles)) {
+            return { ...class_styles, ...styles };
+        }
+        return class_styles;
     };
```

Comment:
Is it possible to `useEffect` for this? :thinking:

Reply:
Sure, I will apply this change in another PR. Thanks.